snap/info.go

```diff
@@ -643,28 +644,69 @@ func (s *Info) CommonDataSaveDir() string {
 	return CommonDataSaveDir(s.InstanceName())
 }
 
+// Get home directories from system-params file.
+// Home directories are returned as a ',' seperated string
+func HomeDirsSysParams(rootDir string) (string, error) {
+
+	ssp, err := sysparams.Open(rootDir)
+	if err != nil {
+		return "", err
+	}
+
+	return ssp.Homedirs, nil
+}
+
 // DataHomeGlob returns the globbing expression for the snap directories in use
-func DataHomeGlob(opts *dirs.SnapDirOptions) string {
+func DataHomeGlob(opts *dirs.SnapDirOptions) []string {
```

Comment:
Maybe it needs to be renamed to `DataHomeGlobs` with the comment updated to describe the new behavior.